### PR TITLE
add functionality for timeout

### PR DESCRIPTION
**Pull request by vinsia**

### CODE REVIEW

**Comment by vinsia** [comment]
@mtth add a timeout to avoid upload and download cost too much time.

File: hdfs/client.py [hunk 1]
```diff
@@ -599,7 +602,7 @@ def wrap(_reader, _chunk_size, _progress):
         for path_tuple in fpath_tuples:
           _upload(path_tuple)
       else:
-        _map_async(n_threads, _upload, fpath_tuples)
+        _map_async(n_threads, _upload, fpath_tuples, timeout=timeout)
```
Comment (mtth) [inline hdfs/client.py]: It is confusing that this feature only works when there is more than one thread. We should keep the two behaviors consistent. It's also not idea that it won't work for Python 2.6, but that's OK given that it's not officially supported anymore.

File: hdfs/client.py [hunk 2]
```diff
@@ -477,7 +478,8 @@ def consumer(_data):
       consumer(data)
 
   def upload(self, hdfs_path, local_path, n_threads=1, temp_dir=None,
-    chunk_size=2 ** 16, progress=None, cleanup=True, **kwargs):
+    chunk_size=2 ** 16, progress=None, cleanup=True,
+             timeout=DEFAULT_TIMEOUT, **kwargs):
```
Comment (mtth) [inline hdfs/client.py]: Can you keep indentation consistent throughout?

File: hdfs/client.py [hunk 3]
```diff
@@ -497,6 +499,7 @@ def upload(self, hdfs_path, local_path, n_threads=1, temp_dir=None,
       completion, it will be called once with `-1` as second argument.
     :param cleanup: Delete any uploaded files if an error occurs during the
       upload.
+    :param timeout: upload timeouts.
```
Comment (mtth) [inline hdfs/client.py]: It would be good to mention the unit and match the form of the other options - can you update?

**Pull request closed without merging**